beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -73,6 +91,9 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
         peer_id: PeerId,
         cx: &mut SyncNetworkContext<T::EthSpec>,
     ) {
+        if let BlockLookupStatus::WaitingOnExecution = self.status() {
+            return;
+        }
```

Comment:
I think we can still request blocks, taking care of limiting how many of those we get